tests/cpp/test_serialization.cc

```diff
@@ -32,7 +32,13 @@ void CompareJSON(Json l, Json r) {
 
     for (auto const& kv : l_obj) {
       ASSERT_NE(r_obj.find(kv.first), r_obj.cend());
-      CompareJSON(l_obj.at(kv.first), r_obj.at(kv.first));
+      if (kv.first == "default_left" || kv.first == "split_conditions") {
+        auto const& l_arr = get<Array const>(l_obj.at(kv.first));
+        auto const& r_arr = get<Array const>(r_obj.at(kv.first));
+        ASSERT_EQ(l_arr.size(), r_arr.size());
+      } else {
+        CompareJSON(l_obj.at(kv.first), r_obj.at(kv.first));
+      }
```

Comment:
This change allows two models to differ in `default_left` and `split_conditions` but still require them to produce idential `loss_chg`. The relaxation is necessary because the test produced two split candidates with identical `loss_chg` and `findex` but different `default_left` and `split_conditions`. The given scenario occurred because there was no missing values for that particular feature.